Backend/dev/test-tool/context-api/server.py

```
@@ -336,6 +342,221 @@ def get_full_context():
+
+
+def get_finance_dashboard(schema):
+    """Full finance dashboard data: accounts, timeline grouped by reference_id, summary stats."""
+    accounts = query(f"""
+        SELECT a.id, a.account_type, a.counterparty_type, a.counterparty_id,
+               a.balance, a.currency, a.status,
+               COALESCE((SELECT SUM(le.amount) FROM {schema}.finance_ledger_entry le WHERE le.to_account_id = a.id AND le.amount > 0), 0) as total_credits,
+               COALESCE((SELECT SUM(ABS(le.amount)) FROM {schema}.finance_ledger_entry le WHERE le.from_account_id = a.id AND le.amount > 0), 0) as total_debits
+        FROM {schema}.finance_account a
+        ORDER BY a.account_type, a.counterparty_type
+    """)
```

Ruff (0.15.10)

[error] Possible SQL injection vector through string-based query construction

(S608)